### PR TITLE
chore: improve documentation of optional jwt property

### PR DESCRIPTION
**Pull request by deekshas8**
## What Has Changed?

Closes: https://github.com/SAP/cloud-sdk-backlog/issues/898

- Add explicit cases where jwt is mandatory
- Add note about approuter session handling in sdk in browser 

## Manual Checks?

- [ ] Check spelling and grammar, e.g., using Grammarly.
- [ ] Verify links still work, e.g., if `id` or the name of a file is changed.
- [ ] Update the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js), e.g., if a feature is added.

### CODE REVIEW

File: docs-js/features/connectivity/destination.mdx [hunk 1]
```diff
@@ -424,7 +424,9 @@ A few of the options were already listed above, but this section gives a compreh
   This is the only mandatory property, all the other parameters are optional.
 - `serviceBindingTransformFn`: A custom transformation function to control how a `Destination` is built from the given `Service`.
 - `jwt`: The JSON Web Token.
-  Crucial in multi-tenant scenarios or for user-dependent authentication flows.
+  The property is mandatory in the following cases:
+  - User-dependent authentication flow is used, e.g., `OAuth2UserTokenExchange`, `OAuth2JWTBearer`, `OAuth2SAMLBearerAssertion`, and `SAMLAssertion`.
+  - Value of `selectionStrategy` is set to `alwaysSubscriber`.
```
Comment (FrankEssenberger) [inline docs-js/features/connectivity/destination.mdx:429]: [pp] Put the same sentence there like in the JS doc in the internal repo

Comment (FrankEssenberger) [inline docs-js/features/connectivity/destination.mdx:429]:
```suggestion
- Multi-tenant scenarios with destinations maintained in the subscriber account. This case is implied if the the  `selectionStrategy` is set to `alwaysSubscriber`.
```

File: docs-js_versioned_docs/version-v2/guides/how-to-use-the-approuter.mdx [hunk 2]
```diff
@@ -25,7 +25,7 @@ You will learn how to secure your application and configure multi-tenancy for pr
 ## SAP Application Router
 
 When we want to enable multi-tenancy for our application, we use the [SAP Application Router](https://www.npmjs.com/package/@sap/approuter).
-The application router’s primary purpose is to be the single entry point of a microservice-based application and act as the application’s reverse proxy.
+The application router's primary purpose is to be the single entry point of a microservice-based application and act as the application's reverse proxy.
```
Comment (FrankEssenberger) [inline docs-js_versioned_docs/version-v2/guides/how-to-use-the-approuter.mdx:28]: [q] Why did the ticks change? Is this the "correct" tick?

**Review by FrankEssenberger — CHANGES_REQUESTED** [review]
I think principal propagation is missing

File: docs-js_versioned_docs/version-v2/features/connectivity/destination.mdx [hunk 3]
```diff
@@ -423,7 +423,9 @@ A few of the options were already listed above, but this section gives a compreh
   This is the only mandatory property, all the other parameters are optional.
 - `serviceBindingTransformFn`: A custom transformation function to control how a `Destination` is built from the given `Service`.
 - `jwt`: The JSON Web Token.
-  Crucial in multi-tenant scenarios or for user-dependent authentication flows.
+  The property is mandatory in the following cases:
+  - User-dependent authentication flows is used, e.g., `OAuth2UserTokenExchange`, `OAuth2JWTBearer`, `OAuth2SAMLBearerAssertion`, and `SAMLAssertion`.
+  - Value of `selectionStrategy` is set to `alwaysSubscriber`.
```
Comment (deekshas8) [inline docs-js_versioned_docs/version-v2/features/connectivity/destination.mdx:428]:
```suggestion
  - Multi-tenant scenarios with destinations maintained in the subscriber account. This case is implied if the the `selectionStrategy` is set to `alwaysSubscriber`.
```

File: docs-js_versioned_docs/version-v2/guides/how-to-use-the-approuter.mdx (hunk 2 above)
Comment (deekshas8) [reply docs-js_versioned_docs/version-v2/guides/how-to-use-the-approuter.mdx:28]: This gave a warning. Also we typically use the apostrophe and not the right single quote.

**Review by FrankEssenberger — APPROVED** [review]

**Pull request merged**